### ISSUE TITLE
Add Ajax support to search select

### ISSUE THREAD

**Issue by calebmeyer** [cerner/terra-core#3241]
# Feature Request

## Description
I would like to be able to load a set of options after the page load for terra form selects. You can do this today if the select requires no input to get 100% of the options, just don't render the select until you have the options in hand.

However, if you want to infinite scroll or search because the list of options is too long, there's not a built-in way to do that.

## Additional Context / Screenshots
We have an issue [PHANALYTIC-16899 - Uplift selects to terra form selects from select2](https://jira2.cerner.com/browse/PHANALYTIC-16899) to convert all our existing select2 usage in HealtheAnalytics, and we can't move forward on it until this is resolved.

Examples:
When adding users to the security of any content (project/data model/schema/feed), we have to use one of the personnel apis to get those users. Those APIs are too slow to hold the page on, so we need to be able to ajax in users. Here's what the select2 looks like:
<img width="857" alt="image" src="https://user-images.githubusercontent.com/3342777/99719781-625f8e80-2a72-11eb-9313-841baafce08f.png">
<img width="861" alt="image" src="https://user-images.githubusercontent.com/3342777/99719991-a9e61a80-2a72-11eb-870c-1a1e2823f074.png">

And with no search
<img width="859" alt="image" src="https://user-images.githubusercontent.com/3342777/99719722-4fe55500-2a72-11eb-9eac-073ef60e861d.png">

On Data Models, we'd like to be able to add a reference to a Data Set, which we call a data set alias. However, many clients have thousands of data sets, so we have to load that select in via ajax:
<img width="1983" alt="image" src="https://user-images.githubusercontent.com/3342777/99720091-d0a45100-2a72-11eb-8651-05f3ff81e3a9.png">
<img width="776" alt="image" src="https://user-images.githubusercontent.com/3342777/99720078-cc783380-2a72-11eb-9b63-5883d1a8eaeb.png">

We made an attempt to convert this one over to terra components, which prompted this issue. Using JSX in a SearchSelect.Option lets us do:
<img width="695" alt="image" src="https://user-images.githubusercontent.com/3342777/99720665-b028c680-2a73-11eb-989c-7bc30464c3a9.png">
<img width="682" alt="image" src="https://user-images.githubusercontent.com/3342777/99720693-b8810180-2a73-11eb-9e55-e3eff6c7b7b3.png">
<img width="672" alt="image" src="https://user-images.githubusercontent.com/3342777/99720721-c20a6980-2a73-11eb-8dbc-a1efd8151e27.png">

but it throws an error in the Javascript console because SearchSelect.Option expects a string, not a node. [Relevant Teams Conversation](https://teams.microsoft.com/l/message/19:3ed4d4d6ffba48138550b9a91a2b50b4@thread.skype/1605807567421?tenantId=fbc493a8-0d24-4454-a815-f4ca58e8c09d&groupId=45f47bfa-38a9-444d-8933-9c20266ae38c&parentMessageId=1605807567421&teamName=Terra&channelName=UX%20Support&createdTime=1605807567421)

## Notes
This is similar to, but not the same as #2369. Implementing the ability to dynamically add options will get us partway to ajax support, but still would present the same problem as above. Ideally terra should provide some indication that an ajax call is in progress and a good way to handle both success and error.

## @ Mentions
@StephenEsser

**Comment by ryanthemanuel** [comment]
There are accessibility concerns with doing the inline search as you've described. We will need to do some exploration on what the design pattern for this type of workflow should be factoring in accessibility, usability, etc.

**Comment by ryanthemanuel** [comment]
Here's the spike where we will look into this: https://github.com/cerner/terra-core/issues/3249
